### PR TITLE
Fix default imports and add a test

### PR DESCRIPTION
**Pull request by keynmol**
*Issue #, if available:*

*Description of changes:*

- Only if the imports are empty, do we add current folder (`.`)
- Small refactoring in SmithyProject
- Fix `recompile(file)` method on SmithyProject

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/main/java/software/amazon/smithy/lsp/SmithyTextDocumentService.java [hunk 1]
```diff
@@ -55,9 +55,9 @@
 
 public class SmithyTextDocumentService implements TextDocumentService {
 
-    private List<CompletionItem> baseCompletions = new ArrayList<CompletionItem>();
+    private final List<CompletionItem> baseCompletions = new ArrayList<CompletionItem>();
     private Optional<LanguageClient> client = Optional.empty();
-    private List<Location> noLocations = Arrays.asList();
+    private final List<Location> noLocations = Collections.emptyList();
```
Comment (keynmol) [inline src/main/java/software/amazon/smithy/lsp/SmithyTextDocumentService.java:60]: Most of the changes in this file can be described by this:
image

But I won't do it again as it goes againt the internal checkstyle 🙃

File: src/main/java/software/amazon/smithy/lsp/ext/SmithyProject.java [hunk 2]
```diff
@@ -65,16 +69,20 @@ private SmithyProject(List<Path> imports, List<File> smithyFiles, List<File> ext
...
         // it's simpler than trying to manage which file was added/removed/closed etc.
-        List<File> newFiles = new ArrayList<File>();
-        newFiles.addAll(onlyExistingFiles(this.smithyFiles));
+        List<File> newFiles = new ArrayList<File>(onlyExistingFiles(this.smithyFiles));
 
         if (file.isFile()) {
             newFiles.add(file);
         }
 
-        return load(this.imports, this.smithyFiles, this.externalJars, this.root);
```
Comment (keynmol) [inline src/main/java/software/amazon/smithy/lsp/ext/SmithyProject.java:77]: This was broken (notice the `this.smithyFiles` even though we need to use `newFiles`)

Found with the test!

File: src/main/java/software/amazon/smithy/lsp/ext/SmithyProject.java [hunk 3]
```diff
@@ -109,7 +117,7 @@ private SmithyProject(List<Path> imports, List<File> smithyFiles, List<File> ext
         List<Path> imports = config.getImports().stream().map(p -> Paths.get(root.getAbsolutePath(), p).normalize())
                 .collect(Collectors.toList());
 
-        if (!imports.contains(Paths.get("."))) {
+        if (imports.isEmpty()) {
```
Comment (keynmol) [inline src/main/java/software/amazon/smithy/lsp/ext/SmithyProject.java:120]: Main change

File: build.gradle [hunk 4]
```diff
@@ -58,7 +58,7 @@ repositories {
 publishing {
     publications {
         mavenJava(MavenPublication) {
-            groupId = "com.disneystreaming.oss"
+            groupId = "com.disneystreaming.smithy"
```
Comment (Baccata) [inline build.gradle:61]: ~alternatively you could use `mill lsp.publishLocal`, which would save us this change. Not bothered though.~ 

nah I'm dumb

File: src/main/java/software/amazon/smithy/lsp/ext/SmithyProject.java [hunk 5]
```diff
@@ -172,21 +180,29 @@ private SmithyProject(List<Path> imports, List<File> smithyFiles, List<File> ext
         return locations;
     }
 
+    private static Boolean isValidSmithyFile(Path file) {
+        String fName = file.getFileName().toString();
+        return fName.endsWith(SMITHY_EXTENSION);
+    }
+
+    private static List<File> walkSmithyFolder(Path path, File root) {
```
Comment (Baccata) [inline src/main/java/software/amazon/smithy/lsp/ext/SmithyProject.java:188]: :+1:

**Review by Baccata — APPROVED** [review]
Noïce

**Pull request merged**